lib/daru/category.rb

```diff
@@ -918,10 +918,28 @@ def modify_category_at pos, category
       end
       old_category = category_from_position pos
       @array[pos] = int_from_cat category
-      @cat_hash[old_category].delete pos
+      @cat_hash[old_category].delete pos unless old_category.nil?
       @cat_hash[category] << pos
     end
 
+    def set pos, category
+      unless categories.include? category
+        raise ArgumentError, "Invalid category #{category}, "\
+          'to add a new category use #add_category'
+      end
+      pos = pos.is_a?(Array) ? pos : [pos]
+      pos.each do |x|
+        if @index.include? x
+          self.[]= x, category
+          next
+        end
+        @array << categories.index(category)
+        @index = @index.add(x)
+        @cat_hash.store(category,[x])
+        self.[]= x, category
```

Comment:
I think this line `@cat_hash.store(category,[x])` will delete the previous value of `@cat_hash[category]` and hence forgetting the previous indexes where the category occurs.

```ruby
[1] pry(main)> a = {cat: [1, 2, 3]}
=> {:cat=>[1, 2, 3]}
[2] pry(main)> a.store(:cat, [4])
=> [4]
[3] pry(main)> a
=> {:cat=>[4]}

```

Reply:
I have removed these lines in next commit because the new index added will be hashed in continuation of the previous index by the`[ ]=` method. So storing an index say 100 in cat_hash after the natural index 2 will be problematic because `[ ]=` will store the index 100 as 3. Removing this line doesn't affect the working.